src/modules/expression/literal/mod.rs

```diff
@@ -27,6 +27,31 @@ fn is_escaped(word: &str, symbol: char) -> bool {
     backslash_count % 2 != 0
 }
 
+fn validate_escape_sequences(meta: &mut ParserMetadata, string_content: &str, tok: Option<Token>) {
+    let mut chars = string_content.chars().peekable();
+    while let Some(c) = chars.next() {
+        if c == '\\' {
+            if let Some(&next_char) = chars.peek() {
+                match next_char {
+                    // Valid escape sequences
+                    'n' | 't' | 'r' | '0' | '{' | '$' | '\'' | '"' | '\\' => {
+                        chars.next(); // consume the valid escape character
+                    }
+                    // Invalid escape sequences
+                    _ => {
+                        let warning_msg = format!("Invalid escape sequence '\\{next_char}'");
+                        let message = Message::new_warn_at_token(meta, tok.clone())
```

Comment:
We don't need to clone this token here

```suggestion
                        let message = Message::new_warn_at_token(meta, tok)
```